### ISSUE TITLE
Qubes Manager: Exiting progress window doesn't cancel associated process

### ISSUE THREAD

**Issue by axon-qubes** [QubesOS/qubes-issues#1479]
To most users, the ability to successfully close a progress bar window by clicking the X in the corner signifies that whatever process was being represented by the progress bar has now been stopped/canceled/aborted. However, this is not how cloning VMs (and probably other operations) in Qubes Manager actually works. Closing the progress bar window just hides it from the user, and the cloning operation continues silently in the background, contrary to most users' expectations.

I suppose the immediate, easy fix would be remove the ability to close the progress bar window(s).

**Comment by unman** [comment]
@andrewdavidwong Can you clarify this? Do you mean that users believe that closing a progress bar will halt the operation? I'm not convinced that is true, but it may be so.
Or did you mean that BEING ABLE to close the progress window makes users believe this.

Rather than removing the ability to close the window, a better option might be to add an explicit Cancel button. That seems to fit better with most UI stuff I've seen.
Alternatively, make closing the progress window a trigger for canceling the operation.

Thoughts? ( We really do need a GUI or UX person to advise here.)

**Comment by andrewdavidwong** [comment]
> Can you clarify this? Do you mean that users believe that closing a progress bar will halt the operation? I'm not convinced that is true, but it may be so.
> Or did you mean that BEING ABLE to close the progress window makes users believe this.
> 
> Rather than removing the ability to close the window, a better option might be to add an explicit Cancel button. That seems to fit better with most UI stuff I've seen.
> Alternatively, make closing the progress window a trigger for canceling the operation.
> 
> Thoughts? ( We really do need a GUI or UX person to advise here.)

Yes. To be precise, it's believe it's probably true, as a generalization, that users expect that *requesting* to close a progress bar, e.g., by pressing "Cancel" or the "X" *and ostensibly being granted that request* by the system, as evidenced by the progress bar disappearing, is supposed to indicate that *the process that the progress bar was tracking* is no longer occurring.

Consider the alternative: A request to close (*not merely hide or minimize*) the progress bar is successfully granted by the system, yet the process that progress bar was tracking continues unabated silently in the background. This is tantamount to the system lying to the user, and it is precisely what happens in Qubes right now.
